leakcanary-android-core/src/main/java/leakcanary/ConfigBuilder.kt

```diff
@@ -0,0 +1,187 @@
+package leakcanary
+
+import shark.AndroidMetadataExtractor
+import shark.AndroidObjectInspectors
+import shark.AndroidReferenceMatchers
+import shark.FilteringLeakingObjectFinder
+import shark.HeapAnalysisSuccess
+import shark.IgnoredReferenceMatcher
+import shark.KeyedWeakReferenceFinder
+import shark.LeakingObjectFinder
+import shark.LibraryLeakReferenceMatcher
+import shark.MetadataExtractor
+import shark.ObjectInspector
+import shark.ReferenceMatcher
+
+class ConfigBuilder(config: LeakCanary.Config) {
+  private var dumpHeap: Boolean = config.dumpHeap
+  private var dumpHeapWhenDebugging: Boolean = config.dumpHeapWhenDebugging
+  private var retainedVisibleThreshold: Int = config.retainedVisibleThreshold
+  private var referenceMatchers: List<ReferenceMatcher> = config.referenceMatchers
+  private var objectInspectors: List<ObjectInspector> = config.objectInspectors
+  private var onHeapAnalyzedListener: OnHeapAnalyzedListener = config.onHeapAnalyzedListener
+  private var metatadaExtractor: MetadataExtractor = config.metatadaExtractor
+  private var computeRetainedHeapSize: Boolean = config.computeRetainedHeapSize
+  private var maxStoredHeapDumps: Int = config.maxStoredHeapDumps
+  private var requestWriteExternalStoragePermission: Boolean = config.requestWriteExternalStoragePermission
+  private var leakingObjectFinder: LeakingObjectFinder = config.leakingObjectFinder
+  private var useExperimentalLeakFinders: Boolean = config.useExperimentalLeakFinders
+
+
+  /**
+   * Whether LeakCanary should dump the heap when enough retained instances are found. This needs
+   * to be true for LeakCanary to work, but sometimes you may want to temporarily disable
+   * LeakCanary (e.g. for a product demo).
+   *
+   * Defaults to true.
+   */
+  fun dumpHeap(value: Boolean) = apply { dumpHeap = value }
```

Comment:
Remove all javadoc and replace with `@see`, so that we don't have to update two places when we update javadoc.

Reply:
Oooh, that's how to do it! I had a gut feeling there should be a better way than copy&paste the docs. Thanks! 😇